pylint/checkers/variables.py

```diff
@@ -979,267 +992,261 @@ def visit_assignname(self, node: nodes.AssignName) -> None:
     def visit_delname(self, node: nodes.DelName) -> None:
         self.visit_name(node)
 
-    # pylint: disable=too-many-branches
     def visit_name(self, node: nodes.Name) -> None:
-        """Check that a name is defined in the current scope"""
         stmt = node.statement()
         if stmt.fromlineno is None:
             # name node from an astroid built from live code, skip
             assert not stmt.root().file.endswith(".py")
             return
 
+        self._undefined_and_used_before_checker(node, stmt)
+        self._loopvar_name(node)
+
+    @utils.check_messages(
+        "undefined-variable", "used-before-assignment", "cell-var-from-loop"
+    )
+    def _undefined_and_used_before_checker(
+        self, node: nodes.Name, stmt: nodes.NodeNG
+    ) -> None:
         frame = stmt.scope()
         start_index = len(self._to_consume) - 1
 
-        undefined_variable_is_enabled = self.linter.is_message_enabled(
+        self._undefined_variable_is_enabled = self.linter.is_message_enabled(
             "undefined-variable"
         )
-        used_before_assignment_is_enabled = self.linter.is_message_enabled(
+        self._used_before_assignment_is_enabled = self.linter.is_message_enabled(
             "used-before-assignment"
         )
```

Comment:
This should probably be moved to `def open()`.

Reply:
Done!